.circleci/copy-junit-reports.js

```diff
@@ -47,9 +48,20 @@ const destDirPath = join(__dirname, '../test-results/jasmine');
 rmSync(destDirPath, {recursive: true, force: true});
 mkdirSync(destDirPath, {recursive: true});
 
+/** Total number of files copied, also used as a index to number copied files. */
+let copiedFileCount = 0;
 // Copy each of the test result files to the central test result directory which CircleCI discovers
 // test results in.
-testResultPaths.forEach((filePath, i) => {
-  const destFilePath = join(destDirPath, `results-${i}.xml`);
-  copyFileSync(filePath, destFilePath);
+testResultPaths.forEach(([xmlFilePath, cacheStatusFilePath]) => {
+  const shortFilePath = xmlFilePath.substr(testLogPath.length + 1);
+  const testResultData = TestResultData.deserializeBinary(readFileSync(cacheStatusFilePath));
+  if (testResultData.getRemotelyCached()) {
+    console.debug(`Skipping copy of ${shortFilePath} as it was a passing remote cache hit`);
```

Comment:
Are they necessarily passing if they are remotely cached? IIUC it can also be cached in a failing (if all inputs are the same with a previously failing run)

Reply:
By default (which we do not set the flag otherwise), bazel will[ rerun a test if "the test previously failed"](https://cs.opensource.google/bazel/bazel/+/master:src/main/java/com/google/devtools/build/lib/analysis/test/TestConfiguration.java;l=112).

The other options of `true` or `false` would cause it to always cache the results no matter what.

Though for our case, even if the test was cached and failed again, we didn't actually run it again so we wouldn't want to upload the result since its not a new run of the test.

Reply:
Ahh, good point! should be good to go